src/qam.lisp

```diff
@@ -15,14 +15,30 @@
 (defgeneric reset-quantum-state (qam)
   (:documentation "Bring all qubits of the quantum abstract machine QAM to the zero state."))
 
-(defgeneric measure (qam q c)
-  (:documentation  "Non-deterministically perform a measurement on the qubit addressed by Q in the quantum abstract machine QAM. Store the bit at the classical bit memory address C. If C is instead NIL, don't store.
+(defgeneric measure (qam q)
+  (:documentation  "Non-deterministically perform a measurement on the qubit addressed by Q in the quantum abstract machine QAM.
 
 Return two values:
 
     1. The resulting QAM.
     2. The measured classical bit."))
 
+(defgeneric measure-and-store (qam q c)
+  (:documentation "Performs a measurement on the qubit addressed by Q in the quantum abstract machine QAM, and stores the measured
+bit in the classical bit addressed by C.
+
+Return two values:
+
+    1. The resulting QAM.
+    2. The measured classical bit."))
+
+(defmethod measure-and-store ((qam quantum-abstract-machine) q c)
```

Comment:
I think we ought to make this a plain old function. Give fewer hooks for the user to hang functionality onto. :)

Do provide a docstring once changed to a `defun`.